### DISCUSSION TITLE
Environment Mesh related utility in MRTK3

### DISCUSSION THREAD

**Discussion by thisisvaze** [microsoft/MixedRealityToolkit-Unity#11228] [Q&A] [answered]
Is it possible to get a custom raycast intersection point with the Environment Mesh in MRTK3?

**Comment by camnewnham** [comment, marked as answer]
I'd suggest the following:
- Ensure that you are creating the environment mesh (you have an `ARMeshManager` under your `ARSessionOrigin`)
- Create a meshprefab that is for a custom layer in your `ARMeshManager`. Ensure this prefab also includes a mesh collider.
- Use `Physics.Raycast` with a mask for your specific layer
